dspace-rest/src/main/java/org/dspace/rest/common/DSpaceObject.java

```diff
@@ -112,4 +114,8 @@ public String getUUID() {
     public void setUUID(String uuid) {
         this.uuid = uuid;
     }
+
+    private String createLink(ServletContext context){
+        return context.getContextPath() + "/" + English.plural(getType()) + "/" + getId();
+    }
```

Comment:
I see what you mean, the link will refer to the ID, and if ID is integer, as opposed to the new UUID, then the link will be limited as new objects will not have old integer ID, but all objects will have uuid. To ensure compatibility, perhaps we would need to leave getLink to return the integerID, but getLinkUUID could use the new identifier. (this is because we don't otherwise have rest versioning, and don't want to break contracts).